pkgs/by-name/ma/matlab/package.nix

```diff
@@ -0,0 +1,242 @@
+{
+  lib,
+  stdenvNoCC,
+  buildFHSEnv,
+  copyDesktopItems,
+  libarchive,
+  makeDesktopItem,
+  writeShellScriptBin,
+
+  fetchFromMPM,
+  matlab-package-manager,
+}:
+
+stdenvNoCC.mkDerivation (self: {
+  pname = "matlab";
+  inherit (self.src) version;
+  products = self.src.matlab-download.products or [ "MATLAB" ];
+
+  matlabBins = [
+    "matlab"
+    "mex"
+    "mlint"
+  ];
+
+  dontUnpack = true;
+  dontConfigure = true;
+  dontBuild = true;
+
+  src = fetchFromMPM {
+    release = "R2024b";
+    update = 1;
+    extra-products = [ ];
+    hash = "sha256-t7+vIk7254mo7dUaRFjZp/qHRpVWnIhy3OEoZfhwqUI=";
+  };
+
+  nativeBuildInputs = [ copyDesktopItems ];
+
+  unwrapped =
+    let
+      unpack =
+        if self.src ? "matlab-download" then
+          # Downloaded using fetchFromMPM: already unpacked
+          { dontUnpack = true; }
+        else
+          # Assume this is an ISO image from requireFile: need to unpack
+          {
+            nativeBuildInputs = [ libarchive ];
+            unpackCmd = "bsdtar -vxf $curSrc";
+            sourceRoot = ".";
+          };
```

Comment:
The unpack phase doesn't apply in-place to `src`; the latter is read-only in the build sandbox. Are you trying to use the files unpacked into the build directory or use `$src` directly?

Reply:
> Thanks! I've pushed some commits which I think address this. Could you have a look?

The unpacking logic is now closed to most other programs.

Still, I guess one of the motivations behind the previous implementation is to avoid unnecessary copying of the large binaries downloaded by matlab-package-manager.

To achieve that, I would suggest keeping `dontUnpack = true` and deligating all the unpacking work to the derivation assigned to `src`. The Matlab install command in the install phase can then use `$src` directly.

Reply:
As for the ISO unpacking logic, the assumption and implementation seem hard to discover if we just buried them in the source code.

How about we drop them and document the overriding in the Nixpkgs Manual with an example of using the ISO image instead? If it is necessary to provide such support in the implementation, we should make it explicit by guarding it with an overridable switch (e.g., `self.isISOSource`)